chore(issue-stream): Remove issue stream component flags from FE

MichaelSun48 opened the pull request:
This PR removes all instances of the feature flags used to rollout the new event stats graph and assignee selector. It has been EA'd for a week and has been in GA since Monday without any issues. 

[Associated options PR](https://github.com/getsentry/sentry-options-automator/pull/1556)

Commit:
Remove new issue stream component flags from FE

snigdhas committed:
chore(issue-platform): Remove the `organizations:issue-platform` flag from the backend (#72071)

Removing the `organizations:issue-platform` feature flag now that it's
been released in self-hosted
(https://github.com/getsentry/self-hosted/pull/2309).

Swatinem committed:
Fix deserialization of Enhancements (#72226)

Serialization using `Enhancements.dumps` was using shorthands (like
`v/value`), whereas the deserialization code did not support those
shorthands previously, but expected the full matcher name (like
`error.value`).

Adding these shorthands makes sure we support ser/de roundtrips for
custom enhancement rules using error matchers.

Fixes #60335

asottile-sentry committed:
ref: fix type for group_states in sentry.event_manager (#72216)

the called function uses a TypedDict so dict[str, int] was not matching

<!-- Describe your PR here. -->

asottile-sentry committed:
ref: fix casts in sentry.event_manager (#72215)

<!-- Describe your PR here. -->

gggritso committed:
feat(insights): Collapse "Insights" sidebar accordion by default (#72156)

Request from Design to, on first page load, collapse the "Insights"
accordion, so it's not as distracting. After that, works as usual, and
stores open/closed state in `localStorage`

k-fish committed:
feat(traces): Add graph for matching spans to traces (#72020)

### Summary
This includes a graph that displays # of cumulative matching spans
according to your search conditions.

Other:
- Small changes to not show 'matching' spans when you have no filters,
instead will show 'total' and not have 'of'

#### Screenshots
![Screenshot 2024-06-04 at 12 38
27 PM](https://github.com/getsentry/sentry/assets/6111995/3698b8e1-052e-47b6-a195-f1547a2a2a7c)

leeandher commented on static/app/components/badge/groupPriority.tsx:
```diff
@@ -99,14 +99,8 @@ export function makeGroupPriorityDropdownOptions({
-        'issue-stream-new-assignee-dropdown-trigger'
-      )}
-    >
+    <StyledTag type={getTagTypeForPriority(priority)} isBigtag>
```
Since this is the only use of `StyledTag` you can remove the `isBigtag` prop and just make the `height` static

MichaelSun48 replied:
Good point! Removing now

leeandher reviewed (commented):
One comment!

Zylphrex committed:
feat(trace-explorer): Add feedback button (#72230)

Commit:
Remove big tag

Zylphrex committed:
feat(trace-explorer): Bump trace explorer to 50 results (#72235)

Since we can't easily paginate, bump the number of results to show more
traces by default.

anonrig committed:
fix: remove appstore connect fields from POST and PUT (#72206)

This removes the fields for POST and PUT on appstore connect integration
and potentially rejecting PUT requests involving AppStore connect
integration.

This change is required to remove AppStore connect from the following
pages:

- https://docs.sentry.io/api/projects/update-a-projects-symbol-source/
- https://docs.sentry.io/api/projects/add-a-symbol-source-to-a-project/

Ref: https://github.com/getsentry/sentry/issues/51994

evanpurkhiser committed:
test(js): Remove `route` from initializeOrg (#72062)

markstory committed:
feat(hybridcloud) Add cached implementation of user.get_many_by_id (#71826)

We make a ton of RPC calls for lists of users. Many of those calls are
simple ID lists. We can cache these quite effectively as the queries are
simple and cache keys can be expired via outboxes.

The rollout of using cache is controlled by an option, and I've added
metrics to observe the effective hit rate of cached/rpc fetches.

Caching decorators were added in #71719

asottile-sentry committed:
ref: improve types for tsdb (#72239)

<!-- Describe your PR here. -->

markstory committed:
chore(hybridcloud) Use cached installation read all the time (#72040)

Remove the rollout option for cached installation reads.

Refs HC-1185

getsentry-bot committed:
Revert "Reapply "ref: upgrade sentry to python 3.12 (#65075)" (#72105)"

This reverts commit 33f55ff555e9dfb6e280758245b30596b935cc9a.

Co-authored-by: asottile-sentry <103459774+asottile-sentry@users.noreply.github.com>

asottile-sentry committed:
ref: fix types for sentry_apps.interaction (#72236)

<!-- Describe your PR here. -->

asottile-sentry committed:
ref: fix typing for relay redis cluster check under test (#72240)

`cluster` is specific to the redis implementation of
`ProjectConfigCache` and otherwise can't safely be accessed on
`projectconfig_cache.backend`

<!-- Describe your PR here. -->

scttcper committed:
ref(ui): Strict config store state (#72174)

evanpurkhiser committed:
test(js): Simplify types in initializeOrg (#72243)

anonrig committed:
fix: use explicit webpack define (#72242)

We need to be explicit when calling define plugin. The proposed change
is more correct than previous one.

scttcper commented on static/app/components/stream/group.tsx:
```diff
@@ -497,40 +496,23 @@ function BaseGroupRow({
-            >
-              {groupCount}
-            </EventCountsWrapper>
+            <EventCountsWrapper leftMargin={space(0)}>{groupCount}</EventCountsWrapper>
```
move leftMargin out of props and into the css since its now static. or remove it entirely since its zero

asottile-sentry committed:
ref: fix types for NodeStore (#72245)

<!-- Describe your PR here. -->

evanpurkhiser committed:
feat(uptime): Add uptime app (#72248)

We'll put stuff here for the upcoming uptime feature

k-fish committed:
fix(traces): Fix empty matches (#72237)

### Summary
The graph wasn't working with empty matches, just have to filter those
out

Abdkhan14 committed:
fix(new-trace): Hiding replay in txn details when in replay details. (#72247)

![Screenshot 2024-06-06 at 1 05
20 PM](https://github.com/getsentry/sentry/assets/60121741/fb20e1e1-d986-40f8-a8b1-83e3b84ae5eb)

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>
Co-authored-by: Jonas <jonas.badalic@sentry.io>

iamrajjoshi committed:
ref(messaging): Use `ExternalProviderEnum.value` (#72186)

This PR is required to fix the tests in
https://github.com/getsentry/getsentry/actions/runs/9389683374/job/25875213718?pr=14211

Because we use the actual Enum class in some of our logic, when I
attempt to use the new copied enum class that i created under the
integration folder, we actually have a mismatch in our code as
`getsentry` will use the newer class while `sentry` won't.

Specifically in this
[line](https://github.com/getsentry/sentry/blob/master/src/sentry/notifications/notificationcontroller.py#L602),
because the defaultdict is keyed with the Enum class (and not the
value), the same enum in a different module will actually not exist in
the dict 😢

We need to use the actual enum value (which is a str), which would be
the same for both classes.

thanks to @ykamo001  for helping out!

Zylphrex committed:
fix(trace-explorer): Do not adjust slice when it overflows (#72229)

We try to adjust the slice so it has a width of 1 but some times it
overflows then gets clipped. This prevents the overflow from happening
in the first place.

anonrig committed:
refactor: simplify webpack.config types (#72246)

Simplifies and removes several types in webpack config.

anonrig committed:
refactor: move ts-fork options to webpack (#72244)

I didn't understand why we defined this in package.json rather than in
webpack.config where it belongs.

Commit:
feat(self-hosted): Remove other categories from stats dropdown for errors only sh (#72154)

![Screenshot 2024-06-05 at 11 37
39 AM](https://github.com/getsentry/sentry/assets/25517925/6efb9118-e94d-45eb-8d62-7960f881010d)

jangjodi committed:
ref(similarity): Add more logs to backfill (#72250)

Add more logs to backfill since it's stopping without any logs

onkardeshpande93 committed:
fix: double tooltips shown on discover export button  (#71501)

Fixes https://github.com/getsentry/sentry/issues/64553

Export button shows the Feature gate tooltip as well as the small export
size tooltip.
Should not show the small export size tooltip if Feature is not
available.

gggritso committed:
chore(insights): Remove `spans-first-ui` flag registration (#72255)

Not in use. In da bin.

schew2381 committed:
chore(api): Publish change team role endpoint (#71953)

I changed the `get_choices` to actually return a tuple of 
`(id, description) e.g. (contributor, Base role that has access to...)`
instead of
`(id, name) e.g. (contributor, Contributor)` so that we could actually
use the function in `serializer.choiceField` and not have to duplicate
the information already stored in `server.py`.

Note that this endpoint's functionality is actually already duplicated
in from [another published
API](https://docs.sentry.io/api/organizations/update-an-organization-members-roles/),
but I'm still choosing to publish it for convenience for our users.

Another thing to note is that some org roles are guaranteed a minimum
team-level role of `admin`. This functionally works, but there is a bug
where you can actually successfully set someone with a minimum role to
`contributor`, and the endpoint will return a 200. It will then store
that `contributor` role, but when in the UI and functionally the person
actually still has their minimum team-level role.
 
Closes ALRT-73

<img width="1483" alt="Screenshot 2024-06-03 at 3 26 28 PM"
src="https://github.com/getsentry/sentry/assets/67301797/775e5e5d-e37e-4bb8-8f2b-990e257ba5da">

sentaur-athena committed:
chore(apis): Update ownership from issues to alerts & notifications (#71190)

After re-orgs some endpoints are better owned by alerts & notifications
vs. issues. I'm not sure if the ownership I'm assigning in this PR is
correct so will get help from both teams.

k-fish committed:
ref(traces): Move traces url and remove breadcrumb (#72141)

### Summary
This changes the traces folder location, updates the url, and removes
the breadcrumb as it's no longer under performance.

schew2381 committed:
nit(api): Switch params to capital ID (#72259)

Per docs team advice and
https://developers.google.com/style/word-list#id

gggritso committed:
chore: Lint Webpack file (#72263)

Bad changes somehow snuck into 72246 but were merged? Unclear, but
blocking deploys.

markstory committed:
chore(features) Move feature defaults for temporary features (#72113)

All feature defaults are now defined when features are registered.
Adding and removing a feature no longer requires changes to `server.py`

Refs HC-1152

snigdhas committed:
chore(issue-platform): Remove the `issue-platform` flag from the frontend (#72072)

FE counterpart to https://github.com/getsentry/sentry/pull/72071

ryan953 committed:
ref(replay): Refactor the replayCompareModal (#72184)

so we can re-use the diff tools inline on another page, outside of any
modal.

Relates to https://github.com/getsentry/sentry/issues/72150

snigdhas committed:
chore(issue-platform): Unregister the issue-platform flag (#72073)

Removing the last bits of the `organizations:issue-platform` feature
flag. The BE and FE uses have been removed in
https://github.com/getsentry/sentry/pull/72071 and
https://github.com/getsentry/sentry/pull/72072.

asottile-sentry committed:
ref: remove unused upload= argument for chart rendering (always True) (#72257)

upload=False was only utilized by tests

this fixes several type errors when LazyServiceWrapper becomes
typechecked

<!-- Describe your PR here. -->

asottile-sentry committed:
ref: require group ids to start merge (#72218)

if this were to return `None` it would crash in `end_merge`

<!-- Describe your PR here. -->

markstory committed:
chore(hybridcloud) Move hacky AlertRule sentryapp logic to serializer (#72128)

These changes echo the changes made in #71956 and #72035 to AlertRule
serialization. Getting this logic out of the endpoint and into the
serializer will be helpful for reducing complexity here in the future.

Refs HC-1024

markstory committed:
fix(api) Correct and expand org members API docs (#72222)

- User is optional as invites don't have users yet
- Add an example including an invite.

Refs #72219

evanpurkhiser committed:
feat(uptime): Add do-nothing uptime results consumer (#72252)

This reads and decodes results from the uptime-results topic and just
logs them

ryan953 committed:
feat(replay): Add Hydration Error Detected search title text (#72179)

Noticed our new hydration error issue type was missing from this enum.
The enum help populate help-docs within the search dropdown.

malwilley committed:
deps(dnd-kit): Upgrade dnd-kit to latest version (#72171)

KevinL10 committed:
feat(discover): track successful discover query result count (#72177)

Adds `discover_search.success` to track % of discover queries return
empty/non-empty results.

KevinL10 committed:
feat(insights): add missing dashboard analytics (#72261)

Adds additional dashboard analytics:
- [x] Filter bar save/cancel
- [x] Delete, duplicate, and edit widget clicks for both Widget Actions
and through "Edit Dashboard"

asottile-sentry committed:
ref: fix types for project_group_stats (#72238)

<!-- Describe your PR here. -->

KevinL10 committed:
feat(analytics): update search query analytics  (#72138)

Adds an `isMultiProject` field to `search.searched` analytics and a new
Amplitude event to track individual key/value filters.

aliu39 committed:
metrics(replay): add a counter for hydrate error breadcrumbs (#72271)

https://github.com/getsentry/sentry/issues/72266
Can refine with tags, extras, later. Org/project is too high cardinality

ryan953 committed:
ref(replay): Inline some props that are used only once (#72274)

Just inlining some stuff that used to be part of a feature flag.

iamrajjoshi committed:
ref(messaging): Updated Integration Types Imports (#72262)

Final PR to update the rest of the integration imports in sentry over to
the new module that I created. I edit 1 `__init__.py` file, plan is to
clean it up completely in a followup pr.

asottile-sentry committed:
ref: improve typing of get_boost_low_volume_projects_sample_rate (#72273)

when the default is a float it never returns None

<!-- Describe your PR here. -->

asottile-sentry committed:
ref: fix typing of bind_nodes (#72275)

my approach in #72055 did not work -- mypy can't figure out the model
field matches the protocol unfortunately

<!-- Describe your PR here. -->

davidenwang reviewed (approved):
lgtm
